diffsync/store/__init__.py

```diff
@@ -188,6 +206,34 @@ def update_or_instantiate(
 
         return obj, created
 
+    def update_or_instantiate_model(self, obj: "DiffSyncModel") -> Tuple["DiffSyncModel", bool]:
+        """Attempt to update an existing object with provided ids/attrs or instantiate obj.
+
+        Args:
+            instance: An instance of the DiffSyncModel to update or create.
+
+        Returns:
+            Provides the existing or new object and whether it was created or not.
+        """
+        model = obj.get_type()
+        ids = obj.get_unique_id()
+        attrs = obj.get_attrs()
+
+        created = False
+        try:
+            obj = self.get(model=model, identifier=ids)
+        except ObjectNotFound:
+            # Add the object to diffsync adapter
```

Comment:
I believe the concept of an adapter is something that the SSoT framework uses, but it is never mentioned in this repository. Possibly:
`# Add the object to the diffsync instance`?